core/01_core.js

```diff
@@ -43,6 +44,7 @@
   const RING_SIZE = 4 * 1024;
   const NO_PROMISE = null; // Alias to null is faster than plain nulls
   const promiseRing = ArrayPrototypeFill(new Array(RING_SIZE), NO_PROMISE);
+  const promiseIdSymbol = SymbolFor("Deno.core.internalPromiseId");
```

Comment:
Non-blocking but as a possible future improvement: you can use a `v8::Private` if you're worried about the symbol leaking out to user land code (it's visible to `Object.getOwnPropertySymbols()` after all.)

Private symbols are like regular symbols except more... er, private. (Great explanation, Ben. +1, would read again.)

Reply:
Is that construct available in JS land?

Reply:
Not yet, but I can add it. You may have seen node's getHiddenValue() and setHiddenValue() - it's the same mechanism.

Reply:
> Not yet, but I can add it. You may have seen node's getHiddenValue() and setHiddenValue() - it's the same mechanism.

I haven't seen it. I guess I'll add a TODO and we can change it once the API is added.